### PR TITLE
Sidebar refactor continuation

### PR DESCRIPTION
**Pull request by fatih-erikli**

### CODE REVIEW

File: src/fontra/views/editor/panel-glyph-search.js [hunk 1]
```diff
@@ -17,6 +19,49 @@ export default class GlyphSearchPanel extends Panel {
     }
   `;
 
+  setup() {
+    this.editorController.glyphsSearch =
```
Comment (justvanrossum) [inline src/fontra/views/editor/panel-glyph-search.js:23]: Why do you assign to editorController here? I believe you meant simple "this".

File: src/fontra/views/editor/panel-glyph-search.js [hunk 2]
```diff
@@ -17,6 +19,49 @@ export default class GlyphSearchPanel extends Panel {
...
+      this.contentElement.querySelector("#glyphs-search");
+    this.editorController.glyphsSearch.glyphMap =
+      this.editorController.fontController.glyphMap;
+    this.editorController.glyphsSearch.addEventListener(
+      "selectedGlyphNameChanged",
+      (event) => {
```
Comment (justvanrossum) [inline src/fontra/views/editor/panel-glyph-search.js:29]: Please factor this again as a method instead of an inline arrow function.

File: src/fontra/views/editor/editor.js [hunk 3]
```diff
@@ -229,7 +229,7 @@ export class EditorController {
       rootSubscriptionPattern[rootKey] = null;
     }
     await this.fontController.subscribeChanges(rootSubscriptionPattern, false);
-    await this.initGlyphsSearch();
+    this.getSidebarPanel("glyph-search").setup();
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:232]: Can't this be done as part of the constructor of the panel? Or as part of the attach method?

File: src/fontra/views/editor/editor.js [hunk 4]
```diff
@@ -1520,7 +1383,7 @@ export class EditorController {
   async toggleUserSettings(onOff) {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:1383]: Do you mean to address the toggle API as a separate task? I'd really like to get rid of this in editor.js.

File: src/fontra/views/editor/editor.js [hunk 5]
```diff
@@ -1520,7 +1383,7 @@ export class EditorController {
...
     if (onOff && !this._didInitUserSettings) {
       this._didInitUserSettings = true;
-      await loaderSpinner(this.initUserSettings());
+      await loaderSpinner(this.getSidebarPanel("user-settings"));
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:1386]: You replaced an init call with a getSidebarPanel() call. That can't be right.

File: src/fontra/views/editor/panel-glyph-search.js (hunk 1 above)
Comment (justvanrossum) [reply src/fontra/views/editor/panel-glyph-search.js:23]: Or else I don't understand what's goin on here. But you should not assign to editorController from outside of editorController itself.

File: src/fontra/views/editor/panel-selection-info.js [hunk 6]
```diff
@@ -2,6 +2,7 @@ import * as html from "/core/unlit.js";
 import { recordChanges } from "/core/change-recorder.js";
 import { rectFromPoints, rectSize, unionRect } from "/core/rectangle.js";
 import { Form } from "/web-components/ui-form.js";
+import { loaderSpinner } from "/core/loader-spinner.js";
```
Comment (justvanrossum) [inline src/fontra/views/editor/panel-selection-info.js:5]: Unused import?

Comment (fatih-erikli) [reply src/fontra/views/editor/panel-selection-info.js]: Deleted

File: src/fontra/views/editor/editor.js [hunk 7]
```diff
@@ -242,111 +241,18 @@ export class EditorController {
...
   }
 
-  async initGlyphsSearch() {
+  initGlyphsSearch() {
     this.glyphsSearch =
       this.getSidebarPanel("glyph-search").contentElement.querySelector(
         "#glyphs-search"
       );
     this.glyphsSearch.glyphMap = this.fontController.glyphMap;
-    this.glyphsSearch.addEventListener("selectedGlyphNameChanged", (event) =>
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:251]: Please undo this change.

File: src/fontra/views/editor/editor.js [hunk 8]
```diff
@@ -562,12 +467,12 @@ export class EditorController {
     this.cleanGlyphsLayers.scaleFactor = 1 / magnification;
   }
 
-  glyphNameChangedCallback(glyphName) {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:565]: And also undo this change.

**Review by justvanrossum — APPROVED** [review]
